mm2src/coins/eth/tron.rs

```diff
@@ -13,12 +19,26 @@ pub enum Network {
     // TODO: Add more networks as needed.
 }
 
-/// Placeholder for a TRON client.
+/// Draft TRON clients structure.
 #[derive(Clone, Debug)]
-pub struct TronClient;
+pub struct TronClients {
+    pub clients: Vec<TronClient>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct TronClient {
+    pub endpoint: String,
+    pub network: Network,
+    #[serde(default)]
+    pub komodo_proxy: bool, // should be true for any net which requires api key
+}
 
 /// Placeholder for TRON fee params.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct TronFeeParams {
     // TODO: Add TRON-specific fields in future steps.
 }
+
+// Helper function to convert TRX to SUN using U256 type
+// Returns None if multiplication would overflow
+pub fn trx_to_eth_sun(trx: u64) -> Option<U256> { trx.checked_mul(ONE_TRX).map(U256::from) }
```

Comment:
why is it `...eth_sun`? what's the `eth` for?

Reply:
U256 is an ethereum dep type
pure trx to sun function would return u64

Reply:
I updated function name https://github.com/KomodoPlatform/komodo-defi-framework/pull/2467/commits/d094dea0879ddc28995616f8dae5fba4bbf8f6a4
I hope it's clearer now